lib/api/actions/functions.php

```diff
@@ -410,119 +411,184 @@ function beans_reset_action( $id ) {
 }
 
 /**
- * Get action.
+ * Get the action's configuration for the given ID and status. Returns `false` if the action is not registered with
+ * Beans.
  *
+ * @since  1.5.0
  * @ignore
+ * @access private
+ *
+ * @param string $id     A unique string used as a reference.
+ * @param string $status Status for which to get the action.
+ *
+ * @return array|bool
  */
 function _beans_get_action( $id, $status ) {
-
 	global $_beans_registered_actions;
 
-	$id = _beans_unique_action_id( $id );
+	$id                 = _beans_unique_action_id( $id );
+	$registered_actions = beans_get( $status, $_beans_registered_actions );
 
-	if ( ! $registered = beans_get( $status, $_beans_registered_actions ) ) {
+	// If the status is empty, return false, as no actions are registered.
+	if ( empty( $registered_actions ) ) {
 		return false;
 	}
 
-	if ( ! $action = beans_get( $id, $registered ) ) {
+	$action = beans_get( $id, $registered_actions );
+	// If the action is null, return false.
+	if ( is_null( $action ) ) {
 		return false;
 	}
 
 	return (array) json_decode( $action );
-
 }
 
 /**
- * Set action.
+ * Store the action's configuration for the given ID and status.
+ *
+ * What happens if the action's configuration is already registered?  If the `$overwrite` flag is set to `true`,
+ * then the new action's configuration is stored, overwriting the previous one. Else, the registered action's
+ * configuration is returned.
  *
+ * @since  1.5.0
  * @ignore
+ * @access private
+ *
+ * @param string $id        A unique string used as a reference.
+ * @param array  $action    The action configuration to store.
+ * @param string $status    Status for which to store the action.
+ * @param bool   $overwrite Optional. When set to `true`, the new action's configuration is stored, overwriting a
+ *                          previously stored configuration (if one exists).
+ *
+ * @return array
  */
-function _beans_set_action( $id, $action, $status, $overwrite = false ) {
-
-	global $_beans_registered_actions;
-
+function _beans_set_action( $id, array $action, $status, $overwrite = false ) {
 	$id = _beans_unique_action_id( $id );
 
-	// Return action which already exist unless overwrite is set to true.
-	if ( ! $overwrite && ( $_action = _beans_get_action( $id, $status ) ) ) {
-		return $_action;
+	// Get the action, if it's already registered.
+	$registered_action = _beans_get_action( $id, $status );
+
+	// If the action is registered and we are not overwriting, return it.
+	if ( true !== $overwrite && ! empty( $registered_action ) ) {
+		return $registered_action;
 	}
 
-	$_beans_registered_actions[ $status ][ $id ] = json_encode( $action );
+	// Let's set (or overwrite) the action.
+	global $_beans_registered_actions;
+	$_beans_registered_actions[ $status ][ $id ] = wp_json_encode( $action );
 
 	return $action;
-
 }
 
 /**
- * Unset action.
+ * Unset the action's configuration for the given ID and status. Returns `false` if there are is no action
```

Comment:
Ignore this function.  It's part of PR #92.  It's shown here as I built this PR off of PR #92's branch.